src/Console/Command/PublishGitlabCommand.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of composer/satis.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Satis\Console\Command;
+
+use Composer\Command\BaseCommand;
+use Composer\Config;
+use Composer\Config\JsonConfigSource;
+use Composer\Json\JsonFile;
+use Composer\Json\JsonValidationException;
+use Composer\Satis\Publisher\GitlabPublisher;
+use Composer\Util\RemoteFilesystem;
+use Seld\JsonLint\ParsingException;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class PublishGitlabCommand extends BaseCommand
+{
+    protected function configure()
+    {
+        $this
+            ->setName('publish-gitlab')
+            ->setDescription('Uploads a given version to Gitlab')
+            ->setDefinition([
+                new InputArgument('file', InputArgument::OPTIONAL, 'Json file to use', './satis.json'),
+                new InputOption('folder', null, InputOption::VALUE_REQUIRED, 'Folder to search for files'),
+                new InputOption('project-id', null, InputOption::VALUE_REQUIRED, 'Gitlab project id'),
+                new InputOption('project-url', null, InputOption::VALUE_REQUIRED, 'Gitlab project url'),
```

Comment:
AFAICT, the command will not work without these settings. So these should not be options (which are optional by definition) but arguments

Reply:
`project-id` and `project-url` changed to required arguments